src/luascript.cpp

```diff
@@ -1177,6 +1177,16 @@ static void pushLoot(lua_State* L, const std::vector<LootBlock>& lootList)
 	}
 }
 
+static void pushTown(lua_State* L, const Town& town)
+{
+	lua_createtable(L, 0, 3);
+	setField(L, "id", town.id);
+	setField(L, "name", town.name);
+	tfs::lua::pushPosition(L, town.templePosition);
+	lua_setfield(L, -2, "templePosition");
+	tfs::lua::setMetatable(L, -1, "Town");
```

Comment:
adding the metatable is impossible, since the table `Town` does not actually exist in the registry `LUA_REGISTRYINDEX` this is because the TFS API registers the metatables with the `registerClass` method in this index

so it will be necessary to search for the table in the global environment and then apply it as a metatable manually

```cpp
	lua_getglobal(L, "Town");
	lua_setmetatable(L, -2);
```